app/models/mixins/aggregation_mixin.rb

```diff
@@ -1,23 +1,22 @@
 module AggregationMixin
   extend ActiveSupport::Concern
   included do
-    virtual_column :aggregate_cpu_speed,       :type => :integer, :uses => :hosts
-    virtual_column :aggregate_cpu_total_cores, :type => :integer, :uses => :hosts
-    virtual_column :aggregate_physical_cpus,   :type => :integer, :uses => :hosts
-    virtual_column :aggregate_memory,          :type => :integer, :uses => :hosts
-    virtual_column :aggregate_vm_cpus,         :type => :integer, :uses => :vms_and_templates
-    virtual_column :aggregate_vm_memory,       :type => :integer, :uses => :vms_and_templates
-    virtual_column :aggregate_disk_capacity,   :type => :integer, :uses => :hosts
+    virtual_aggregate :aggregate_vm_memory, :vm_hardwares, :sum, :memory_mb
+    virtual_aggregate :aggregate_vm_cpus, :vm_hardwares, :sum, :cpu_sockets
+    virtual_aggregate :aggregate_physical_cpus, :host_hardwares, :sum, :cpu_sockets
+    virtual_aggregate :aggregate_disk_capacity, :host_hardwares, :sum, :disk_capacity
+    virtual_aggregate :aggregate_memory, :host_hardwares, :sum, :memory_mb
+    virtual_aggregate :aggregate_cpu_total_cores, :host_hardwares, :sum, :cpu_total_cores
+    virtual_aggregate :aggregate_cpu_speed, :host_hardwares, :sum, :aggregate_cpu_speed
 
-    alias_method :all_hosts,              :hosts
-    alias_method :all_host_ids,           :host_ids
-    alias_method :all_vms_and_templates,  :vms_and_templates
-    alias_method :all_vm_or_template_ids, :vm_or_template_ids
-    alias_method :all_vms,                :vms
-    alias_method :all_vm_ids,             :vm_ids
-    alias_method :all_miq_templates,      :miq_templates
-    alias_method :all_miq_template_ids,   :miq_template_ids
+    def aggregate_hardware(from, field, targets = nil)
+      from      = from.to_s.singularize
+      select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
```

Comment:
I think this line can be deleted.

I see this `aggregate_hardware` method (not the one in the similarly named `app/models/service/aggregation.rb` that has my fingerprints all over it...) only still being used in 5 places between two files:

https://github.com/ManageIQ/manageiq/blob/2c59fffe4ddf656a77c04f225d3f04ba652989b3/app/models/container_project.rb#L78-L88

https://github.com/ManageIQ/manageiq/blob/2c59fffe4ddf656a77c04f225d3f04ba652989b3/app/models/manageiq/providers/container_manager.rb#L74-L80

None of which actually pass `:aggregate_cpu_speed` to this method, so the `select` will always just be `field`.

Reply:
Also, seems like `:computer_systems` is now the only value ever passed to `aggregate_hardware`, so `from` kinda seems like a pointless argument to pass.

Reply:
oh yea. I can see more methods being replaced with a virtual totals call

Reply:
This is less about "let's use more `virtual_totals`" and more that this line:

```ruby
targets ||= send("all_#{from}_ids")
```

Is always going to be evaluated as:

```ruby
targets ||= send(:all_computer_systems_ids)
```

So why not just hard code it and drop the arg if we are already in here refactoring.

Reply:
| None of which actually pass :aggregate_cpu_speed to this method

I mean, we are testing passing aggregate_cpu_speed to this method on hosts, right?  https://github.com/ManageIQ/manageiq/blob/master/spec/models/mixins/aggregation_mixin_spec.rb#L118

Reply:
thinking about this again, can we dump all calls to `aggregate_hardware`?

if there is only 1 use case, then do we just need to solve that one?

Reply:
from @d-m-u 
```ruby
def aggregate_cpu_total_cores
  Hardware.where(:computer_system_id => computer_systems).sum(:cpu_total_cores)
end
```